test/viewport.js

```diff
@@ -276,6 +278,282 @@
+                "Viewport fit a tall image horizontally."
+            );
+            start();
+        };
+        viewer.addHandler('open', openHandler);
+        viewer.open(TALL_PATH);
+    });
+
+    asyncTest('fitVertically', function(){
+        var openHandler = function(event) {
+            viewer.removeHandler('open', openHandler);
+            var viewport = viewer.viewport;
+            viewport.fitVertically(true);
+            viewport.update();
+            propEqual(
+                viewport.getBounds(),
+                new OpenSeadragon.Rect(0,0,0.25,0.25),
+                "Viewport fit a wide image vertically."
+            );
+            start();
+        };
+        viewer.addHandler('open', openHandler);
+        viewer.open(WIDE_PATH);
+    });
+
+    asyncTest('panBy', function(){
+        var openHandler = function(event) {
+            viewer.removeHandler('open', openHandler);
+            var viewport = viewer.viewport;
+
+            for (var i = 0; i < testPoints.length; i++){
+                var expected = viewport.getCenter().plus(testPoints[i]);
+                viewport.panBy(testPoints[i], true);
+                viewport.update(); // need to call this even with immediately=true
+                propEqual(
+                    viewport.getCenter(),
+                    expected,
+                    "Panned by the correct amount."
+                );
+            }
+
+            start();
+        };
+        viewer.addHandler('open', openHandler);
+        viewer.open(DZI_PATH);
+    });
+
+    asyncTest('panTo', function(){
+        var openHandler = function(event) {
+            viewer.removeHandler('open', openHandler);
+            var viewport = viewer.viewport;
+
+            for (var i = 0; i < testPoints.length; i++){
+                viewport.panTo(testPoints[i], true);
+                viewport.update(); // need to call this even with immediately=true
+                propEqual(
+                    viewport.getCenter(),
+                    testPoints[i],
+                    "Panned to the correct location."
+                );
+            }
+
+            start();
+        };
+        viewer.addHandler('open', openHandler);
+        viewer.open(DZI_PATH);
+    });
+
+    asyncTest('zoomBy', function(){
+        var openHandler = function(event) {
+            viewer.removeHandler('open', openHandler);
+            var viewport = viewer.viewport;
+
+            for (var i = 0; i < testZoomLevels.length; i++){
+                viewport.zoomBy(testZoomLevels[i], null, true);
+                viewport.update(); // need to call this even with immediately=true
+                propEqual(
+                    viewport.getZoom(),
+                    testZoomLevels[i],
+                    "Zoomed by the correct amount."
+                );
+
+                // now use a ref point
+                viewport.zoomBy(testZoomLevels[i], testPoints[i], true);
```

Comment:
To really test reference points, you'd want to make sure that it panned to the right location while zooming. The math on that might be tricky... this might be a place where keeping a list of known good values is good enough.